### PR TITLE
making sure req and res are associted to the domain

### PR DESCRIPTION
**Pull request by dlozlla**

### CODE REVIEW

**Comment by palmerabollo** [comment]
Well spotted, man. Not sure, but in `cleanDomain` we might need to add the following lines in order to clean the domain properly (see CCB):

``` javascript
requestDomain.remove(req);
requestDomain.remove(res);
```

File: lib/domain.js [hunk 1]
```diff
@@ -33,6 +33,9 @@ module.exports = function(logger) {
 
   return function domainMiddleware(req, res, next) {
     var requestDomain = domain.create();
+    //make sure req and res are associated to the newly created domain
```
Comment (palmerabollo) [inline lib/domain.js]: Minor: I'd include a link to [domain explicit binding](https://nodejs.org/api/domain.html#domain_explicit_binding) to describe "why" and not "what" the code does.

Comment (jlorgal) [inline lib/domain.js]: I would comment:

```
// Create a separate domain for each request by explicit binding of request and response
// See (https://nodejs.org/api/domain.html#domain_explicit_binding)
var requestDomain = domain.create();
requestDomain.add(req);
requestDomain.add(res);
```

**Comment by jlorgal** [comment]
@palmerabollo: According to example at https://nodejs.org/api/domain.html#domain_explicit_binding, it is not necessary to remove the explicit binding of request and response. In fact, we destroy the domain.

However, we can live quieter if we add these lines just before requestDomain.exit() line.

File: lib/domain.js [hunk 2]
```diff
@@ -33,6 +33,10 @@ module.exports = function(logger) {
 
   return function domainMiddleware(req, res, next) {
     var requestDomain = domain.create();
+    //we need to explicitly bind req and res to the newly created domain
```
Comment (jlorgal) [inline lib/domain.js]: Format comment to add space after // and to start with upper case:
// We ...
// See ...

**Comment by jlorgal** [comment]
LGTM

**Pull request merged**